framework/transport/ieee1905_transport/ieee1905_transport_broker.cpp

```diff
@@ -0,0 +1,355 @@
+            m_type_to_soc[msg_type.value].erase(sd);
+        }
+    }
+
+    LOG(INFO) << "FD (" << sd->getSocketFd() << ") "
+              << std::string(subscribe ? "subscribed to" : "unsubscribed from")
+              << " the following types: " << types_stream.str();
+
+    LOG(DEBUG) << "FD (" << sd->getSocketFd() << ") subscriptions: " << std::hex
+               << m_soc_to_type[sd] << std::dec;
+
+    return true;
+}
+
+bool BrokerServer::socket_connected(std::shared_ptr<SocketServer> sd)
+{
+    // Accept the connection
+    auto new_socket = std::shared_ptr<Socket>(sd->acceptConnections());
+
+    // Check for errors
+    const auto error_msg = sd->getError();
+    if ((!new_socket) || (!error_msg.empty())) {
+        LOG(ERROR) << "Socket error: " << error_msg;
+        return false;
+    }
+
+    if (!sd->getUdsPath().empty()) {
+        LOG(DEBUG) << "New connection on " << sd->getUdsPath()
+                   << " fd = " << uintptr_t(new_socket->getSocketFd());
+    } else {
+        LOG(DEBUG) << "New connection from ip = " << new_socket->getPeerIP()
+                   << " port = " << new_socket->getPeerPort()
+                   << " fd = " << uint64_t(new_socket->getSocketFd());
+    }
+
+    // Socket event handlers
+    SocketEventLoop::EventHandlers socket_handlers{
+        // Handle incoming data
+        .on_read =
+            [&](SocketEventLoop::EventType socket, SocketEventLoop::EventLoopType &loop) {
+                if (!handle_msg(socket)) {
+                    // NOTE: Do NOT stop the broker on parsing errors...
+                }
+
+                return true;
+            },
+
+        // Not implemented
+        .on_write   = nullptr,
+        .on_timeout = nullptr,
+
+        // Remove the socket on disconnections or errors
+        .on_disconnect =
+            [&](SocketEventLoop::EventType socket, SocketEventLoop::EventLoopType &loop) {
+                if (!socket_disconnected(socket)) {
+                    // NOTE: Do NOT stop the broker on errors...
+                }
+
+                return true;
+            },
+        .on_error =
+            [&](SocketEventLoop::EventType socket, SocketEventLoop::EventLoopType &loop) {
+                if (!socket_disconnected(socket)) {
+                    // NOTE: Do NOT stop the broker on errors...
+                }
+
+                return true;
+            },
+    };
+
+    // Add the newly accepted socket into the poll
+    if (!add_event(new_socket, socket_handlers)) {
+        LOG(ERROR) << "Failed adding new socket into the poll!";
+        return false;
+    }
+
+    return true;
+}
+
+bool BrokerServer::socket_disconnected(std::shared_ptr<Socket> sd)
+{
+    LOG(DEBUG) << "Socket disconnected: FD(" << sd->getSocketFd() << ")";
+
+    // Delete the Socket from the list of types subscriptions
+    for (auto &type : m_soc_to_type[sd]) {
+        m_type_to_soc[type].erase(sd);
+    }
```

Comment:
I think it's a bit pointless to keep the inverse map just for this purpose. You can also do something like:
```suggestion
    // Delete the Socket from the list of types subscriptions
    auto it = std::find_if(m_type_to_soc.begin(), m_type_to_soc.end(), [](pair) { return pair.second == sd; });
    for (; it != m_type_to_soc.end(); it++ ) {
        m_type_to_soc[it->first].erase(sd);
    }
```

Oh well...